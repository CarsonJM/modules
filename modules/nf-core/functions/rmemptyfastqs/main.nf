// create a function that filters channels to remove empty fastq files
def rmEmptyFastQs(ch_fastqs, print_output) {
    def ch_nonempty_fastqs = ch_fastqs
        .filter { meta, fastq ->
                if ( meta.single_end ) {
                    try {
                        fastq.countFastq(limit: 10) > 1
                    } catch (java.util.zip.ZipException e) {
                        log.warn "[HoffLab/phageannotator]: ${fastq} is not in GZIP format, this is likely because it was cleaned with --remove_intermediate_files"
                        true
                    } catch (EOFException) {
                        log.warn "[HoffLab/phageannotator]: ${fastq} has an EOFException, this is likely an empty gzipped file."
                    }
                } else {
                    try {
                        fastq[0].countLines( limit: 10 ) > 1
                    } catch (java.util.zip.ZipException e) {
                        log.warn "[HoffLab/phageannotator]: ${fastq} is not in GZIP format, this is likely because it was cleaned with --remove_intermediate_files"
                        true
                    } catch (EOFException) {
                        log.warn "[HoffLab/phageannotator]: ${fastq[0]} has an EOFException, this is likely an empty gzipped file."
                    }
                }
            }

    if (print_output) {
        ch_nonempty_fastqs.view()
    }
    return ch_nonempty_fastqs
}
