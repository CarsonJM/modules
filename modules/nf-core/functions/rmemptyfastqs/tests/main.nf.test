nextflow_function {

    name "Test rmEmptyFastQs"
    script "../main.nf"


    test("paired end + cleaned.fasta.gz") {
        function "rmEmptyFastQs"
        when {
            function {
            """
            input[0] = Channel.of(
                [
                    [ id:'nonempty', single_end:false], // meta map
                    [
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true),
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_2.fastq.gz', checkifExists: true)
                    ]
                ],
                [
                    [ id:'cleaned', single_end:true ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/sarscov2/illumina/fastq/cleaned.fastq.gz", checkIfExists: true)
                ]
            )
            input[1] = true
            """
            }
        }
        then {
            assertAll (
                { assert function.success },
                { assert snapshot(function.stdout).match() }
            )
        }
    }

    test("single end + empty.fastq.gz") {
        function "rmEmptyFastQs"
        when {
            function {
            """
            input[0] = Channel.of(
                [
                    [ id:'nonempty', single_end: true], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/fastq/test_1.fastq.gz', checkifExists: true)
                ],
                [
                    [ id:'empty', single_end:true ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/sarscov2/illumina/fastq/empty.fastq.gz", checkIfExists: true)
                ]
            )
            input[1] = true
            """
            }
        }
        then {
            assertAll (
                { assert function.success },
                { assert snapshot(function.stdout).match() }
            )
        }
    }
}