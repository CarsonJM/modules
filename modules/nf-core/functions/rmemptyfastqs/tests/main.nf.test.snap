{
    "paired end + cleaned.fasta.gz": {
        "content": [
            [
                "[[id:nonempty, single_end:false], [/nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz, /nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz]]",
                "[[id:cleaned, single_end:true], /mmfs1/gscratch/pedslabs_hoffman/carsonjm/nf-microbe/modules/assets/test-datasets/genomics/sarscov2/illumina/fastq/cleaned.fastq.gz]"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T08:13:09.384239509"
    },
    "single end + empty.fastq.gz": {
        "content": [
            [
                "[[id:nonempty, single_end:true], /nf-core/test-datasets/modules/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz]"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-02T08:13:15.124346278"
    }
}